### DISCUSSION TITLE
How to provider fallback render for missing component

### DISCUSSION THREAD

**Discussion by liaoyeguo** [mdx-js/mdx#2170] [Q&A] [answered]
When we use a component inside mdx file and it is not inject by MDXProvider,  an error will be throw and page crash. Is there a way to provider a fallback render method or add error boundary for this missing component? 

```mdx
This component is not pass to MDXProvider

<Button />
```

```bash
Uncaught Error: Expected component `Button` to be defined: you likely forgot to import, pass, or provide it.
```

**Comment by wooorm** [comment, marked as answer]
The framework you use likely has support for this. Assuming React, it has error boundaries: https://reactjs.org/docs/error-boundaries.html.
